src/stats/stats.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_STATS_H
+#define BITCOIN_STATS_H
+
+#include "amount.h"
+#include "scheduler.h"
+#include "stats/stats_mempool.h"
+
+#include <atomic>
+#include <stdlib.h>
+
+#include <boost/signals2/signal.hpp>
+
+// Class that manages various types of statistics and its memory consumption
+class CStats
+{
+private:
+    static CStats* sharedInstance; //!< singleton instance
+    std::atomic<bool> statsEnabled;
```

Comment:
Why have this as a separate variable when it seems to just be true when maxStatsMemory != 0. Maybe eliminate this or else add comments documenting the intended interfaction between the two variables.